store-collected-images/src/popup.js

```diff
@@ -0,0 +1,112 @@
+"use strict";
+
+import {getFileStorage} from 'idb-file-storage/src/idb-file-storage';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import uuidV4 from 'uuid/v4';
+
+import preventWindowDragAndDrop from './utils/handle-window-drag-and-drop';
+import {saveCollectedBlobs} from './utils/image-store';
+
+class Popup extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      collectedBlobs: [],
+      lastMessage: undefined,
+    };
+
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick(ev) {
+    if (!this.refs.collectionName.value) {
+      this.setState({
+        lastMessage: {text: "The collection name is mandatory.", type: "error"},
+      });
+
+      setTimeout(() => {
+        this.setState({lastError: undefined});
+      }, 2000);
+
+      return;
+    }
+
+    saveCollectedBlobs(this.refs.collectionName.value, this.state.collectedBlobs)
+      .then(() => {
+        this.setState({
+          lastMessage: {text: "All the collected images have been saved", type: "success"},
+          collectedBlobs: [],
+        });
+
+        setTimeout(() => {
```

Comment:
I'm not sure what these setTimeouts are doing, but I'll take your word for it :-).